include/wx/generic/grideditors.h

```diff
@@ -373,6 +373,45 @@ class WXDLLIMPEXP_ADV wxGridCellAutoWrapStringEditor : public wxGridCellTextEdit
     wxDECLARE_NO_COPY_CLASS(wxGridCellAutoWrapStringEditor);
 };
 
+#if wxUSE_DATEPICKCTRL
+
+class WXDLLIMPEXP_ADV wxGridCellDateEditor : public wxGridCellEditor
+{
+public:
+    explicit wxGridCellDateEditor();
+
+    virtual void Create(wxWindow* parent,
+                        wxWindowID id,
+                        wxEvtHandler* evtHandler) wxOVERRIDE;
+
+    virtual void SetSize(const wxRect& rect) wxOVERRIDE;
+
+    // virtual void PaintBackground(wxDC& dc,
+    //                              const wxRect& rectCell,
+    //                              const wxGridCellAttr& attr) wxOVERRIDE;
+
+    // virtual bool IsAcceptedKey(wxKeyEvent& event) wxOVERRIDE;
+    virtual void BeginEdit(int row, int col, wxGrid* grid) wxOVERRIDE;
+    virtual bool EndEdit(int row, int col, const wxGrid* grid,
+                         const wxString& oldval, wxString *newval) wxOVERRIDE;
+    virtual void ApplyEdit(int row, int col, wxGrid* grid) wxOVERRIDE;
+
+    virtual void Reset() wxOVERRIDE;
+    // virtual void StartingKey(wxKeyEvent& event) wxOVERRIDE;
+    // virtual void HandleReturn(wxKeyEvent& event) wxOVERRIDE;
+
+    virtual wxGridCellEditor *Clone() const wxOVERRIDE;
+
+    virtual wxString GetValue() const wxOVERRIDE;
+
+private:
+    wxDateTime m_value;
```

Comment:
It would be nice to add `DatePicker()` accessor hiding the `static_cast<wxDPCtrl*>(m_control)`, as it's done for e.g. `Combo()` accessor in `wxGridCellChoiceEditor` and then use it in the implementation.